yt_dlp/extractor/duoplay.py

```diff
@@ -0,0 +1,107 @@
+            'id': '17_14',
+            'ext': 'mp4',
+            'title': 'Ühikarotid',
+            'thumbnail': r're:https://.+\.jpg(?:\?c=\d+)?$',
+            'description': 'md5:4719b418e058c209def41d48b601276e',
+            'upload_date': '20100916',
+            'timestamp': 1284661800,
+            'series': 'Ühikarotid',
+            'series_id': '17',
+            'season': 'Season 2',
+            'season_number': 2,
+            'episode': None,
+            'episode_number': None,
+            'episode_id': 14,
+        },
+    }, {
+        'note': 'Movie',
+        'url': 'https://duoplay.ee/4325/naljamangud',
+        'md5': '2b0bcac4159a08b1844c2bfde06b1199',
+        'info_dict': {
+            'id': '4325',
+            'ext': 'mp4',
+            'title': 'Näljamängud',
+            'thumbnail': r're:https://.+\.jpg(?:\?c=\d+)?$',
+            'description': 'md5:fb35f5eb2ff46cdb82e4d5fbe7b49a13',
+            'cast': ['Jennifer Lawrence', 'Josh Hutcherson', 'Liam Hemsworth'],
+            'upload_date': '20231109',
+            'timestamp': 1699552800,
+            'series': 'Näljamängud',
+            'series_id': '4325',
+            'season': 'Season 1',
+            'season_number': 1,
+            'episode': 'Episode 0',
+            'episode_number': 0,
+            'episode_id': 0,
+        },
+    }]
+
+    def _real_extract(self, url):
+        telecast_id, episode = self._match_valid_url(url).group('id', 'ep')
+        video_id = join_nonempty(telecast_id, episode, delim='_')
+        webpage = self._download_webpage(url, video_id)
+        video_player = try_call(lambda: extract_attributes(
+            get_element_text_and_html_by_tag('video-player', webpage)[1]))
+        if not video_player or not video_player.get('manifest-url'):
+            self.raise_no_formats('No video found', expected=True)
+
+        episode_attr = self._parse_json(video_player.get(':episode') or '', video_id, fatal=False) or {}
+
+        return {
+            'id': video_id,
+            'title': episode_attr.get('title') if episode_attr.get('category') == 'movies' else (
+                traverse_obj(episode_attr, 'subtitle', ('episode_nr', {lambda x: f'Episode {x}' if x else None}), 'title')),
+            'formats': self._extract_m3u8_formats(video_player['manifest-url'], video_id, 'mp4'),
+            **traverse_obj(episode_attr, {
+                'description': 'synopsis',
+                'thumbnail': ('images', 'original'),
+                'timestamp': ('airtime', {lambda x: unified_timestamp(x + ' +0200')}),
+                'series': 'title',
+                'series_id': ('telecast_id', {str_or_none}),
+                'season_number': ('season_id', {int_or_none}),
+                'episode': 'subtitle',
+                'episode_number': ('episode_nr', {int_or_none}),
+                'episode_id': ('episode_id', {int_or_none}),
+                'cast': ('cast', {lambda x: x.split(', ')}),
+            }),
```

Comment:
No longer include series related data in movies:
```suggestion
            'id': video_id,
            'formats': self._extract_m3u8_formats(video_player['manifest-url'], video_id, 'mp4'),
            **traverse_obj(episode_attr, {
                'title': 'title',
                'description': 'synopsis',
                'thumbnail': ('images', 'original'),
                'timestamp': ('airtime', {lambda x: unified_timestamp(x + ' +0200')}),
                'cast': ('cast', {lambda x: x.split(', ')}),
            }),
            **(traverse_obj(episode_attr, {
                'title': (None, ('subtitle', ('episode_nr', {lambda x: f'Episode {x}' if x else None}))),
                'series': 'title',
                'series_id': ('telecast_id', {str_or_none}),
                'season_number': ('season_id', {int_or_none}),
                'episode': 'subtitle',
                'episode_number': ('episode_nr', {int_or_none}),
                'episode_id': ('episode_id', {int_or_none}),
            }, get_all=False) if episode_attr.get('category') != 'movies' else {}),
```